src/SmoFacade/Listener.cs

```diff
@@ -128,11 +129,21 @@ private static string AgListenerName(string dataSource)
       return dotIndex >= 0 ? dataSource.Remove(dotIndex) : dataSource;
     }
 
-    private static Server AgListenerNameToServer(ref SqlConnectionStringBuilder connBuilder, string agInstanceName,
-      string credentailName)
+    private static Server AgListenerNameToServer(ref SqlConnectionStringBuilder connBuilder, string agInstanceName)
     {
-      connBuilder.DataSource = Dns.GetHostEntry(agInstanceName).HostName;
-      return new Server(connBuilder.ToString(), credentailName);
+      try {
+        connBuilder.DataSource = Dns.GetHostEntry(agInstanceName).HostName;
+      }
+      catch(Exception) {
+        // NamedInstances: chop instance name, resolve DNS, slap instance name back on!
+        var parts = agInstanceName.Split('\\');

```

Comment:
We should just do this from the get go to determine if it's a named instance. 

What are the limitations on SQL hostnames and instance names?
https://docs.microsoft.com/en-us/previous-versions/sql/sql-server-2008-r2/ms143531(v=sql.105)?redirectedfrom=MSDN
https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#:~:text=The%20maximum%20length%20of%20the,from%20the%20NETBIOS%20host%20name.

They can't have a backslash in the name, so we should be good start with this.

Reply:
will do.
I figured FDS names wouldn't have a `\`, but its reassuring that msft restricts it for all named instances

Reply:
@Kaneraz - The problem with this actually is that `\` is not restricted from a servername, only a Named instance.
If I `Split` on backslash and it's a servername, what would happen?  

The exception control flow works nicely to only split the named instance if DNS doesn't match the first attempt

Reply:
I read it the other way, that our servername can't have a `\` in it.

Reply:
gotcha, I guess neither can have `\` in the name